sync/src/types/mod.rs

```diff
@@ -249,11 +249,24 @@ impl HeadersSyncController {
     }
 }
 
+#[derive(Clone, Copy, Eq, PartialEq, Debug)]
+pub enum BinaryVersion {
```

Comment:
I don't think we need a new enum here, just add a new field to struct `PeerFlags`, which can reduce a large number of fn params changes:
```
pub struct PeerFlags {
    pub is_outbound: bool,
    pub is_protect: bool,
    pub is_whitelist: bool,
    pub is_2021edition: bool,
}
```

Reply:
Modification is complete, please review again